packages/circle-demo-webapp/app/components/TransactionDetails/TransactionDetails.tsx

```diff
@@ -9,8 +11,36 @@ export interface TransactionDetailsProps {
   transaction: Transaction;
 }
 
+export interface ActiveTransactionDetailsProps {
+  /** The on-chain transaction */
+  transactionId: string;
+  loadTransactionById: (transactionId: string) => Promise<Transaction>;
+}
+
 /** The details of an on-chain transaction */
-export function TransactionDetails({ transaction }: TransactionDetailsProps) {
+export function TransactionDetails(
+  props: TransactionDetailsProps | ActiveTransactionDetailsProps,
```

Comment:
I believe the number of linting errors in this file is already a strong indicator that something is seriously wrong with this component.

Reply:
discussion open